cli/fmt.rs

```diff
@@ -173,6 +194,50 @@ fn format_stdin(check: bool) -> Result<(), ErrBox> {
   Ok(())
 }
 
+async fn run_parallelized<F>(
+  file_paths: Vec<PathBuf>,
+  f: F,
+) -> Result<(), ErrBox>
+where
+  F: FnOnce(PathBuf) -> Result<(), ErrBox> + Send + 'static + Clone,
+{
+  let handles = file_paths.iter().map(|file_path| {
+    let f = f.clone();
+    let file_path = file_path.clone();
+    tokio::task::spawn_blocking(move || f(file_path))
+  });
+  let join_results = futures::future::join_all(handles).await;
+
+  // find the tasks that panicked and let the user know which files
+  let panic_file_paths = join_results
+    .iter()
+    .enumerate()
+    .filter_map(|(i, join_result)| {
+      join_result
+        .as_ref()
+        .err()
+        .map(|_| file_paths[i].to_string_lossy())
+    })
+    .collect::<Vec<_>>();
+  if !panic_file_paths.is_empty() {
+    panic!("Panic formatting: {}", panic_file_paths.join(", "))
+  }
```

Comment:
So I ended up just adding this code so we can take a look, but let me know if I should revert it/scale it back/make any improvements.

Right now the output looks like this when there's a panic (not ideal, because two panics on different threads are outputted to the console simultaneously, but at least with this it shows the two filenames at the bottom that the panics happened on):

```
thread 'tokio-runtime-worker' panicked at 'not implemented: reparse_expr_as_pat, pat_ty = BindingPat, expr = Arrow(ArrowExpr { span: Span { lo: BytePos(1276817), hi: BytePos(1276901), ctxt: #0 }, params: [Ident(Ident { span: Span { lo: BytePos(1276817), hi: BytePos(1276818), ctxt: #0 }, sym: Atom('t' type=inline), type_ann: None, optional: false })], body: Expr(Bin(BinExpr { span: Span { lo: BytePos(1276822), hi: BytePos(1276900), ctxt: #0 }, op: "||", left: Call(CallExpr { span: Span { lo: BytePos(1276822), hi: BytePos(1276854), ctxt: #0 }, callee: Expr(Ident(Ident { span: Span { lo: BytePos(1276822), hi: BytePos(1276840), ctxt: #0 }, sym: Atom('areTypesComparable' type=dynamic), type_ann: None, optional: false })), args: [ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276841), hi: BytePos(1276842), ctxt: #0 }, sym: Atom('t' type=inline), type_ann: None, optional: false }) }, ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276844), hi: BytePos(1276853), ctxt: #0 }, sym: Atom('valueType' type=dynamic), type_ann: None, optional: false }) }], type_args: None }), right: Call(CallExpr { span: Span { lo: BytePos(1276858), hi: BytePos(1276900), ctxt: #0 }, callee: Expr(Ident(Ident { span: Span { lo: BytePos(1276858), hi: BytePos(1276886), ctxt: #0 }, sym: Atom('isCoercibleUnthread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', C:\Users\david\.cargo\registry\src\github.com-1ecc6299db9ec823\dprint-plugin-typescript-0.13.0\src\parser.rs:1299:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
derDoubleEquals' type=dynamic), type_ann: None, optional: false })), args: [ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276887), hi: BytePos(1276888), ctxt: #0 }, sym: Atom('t' type=inline), type_ann: None, optional: false }) }, ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276890), hi: BytePos(1276899), ctxt: #0 }, sym: Atom('valueType' type=dynamic), type_ann: None, optional: false }) }], type_args: None }) })), is_async: false, is_generator: false, type_params: None, return_type: None })', <::std::macros::panic macros>:5:6
thread 'main' panicked at 'Panic formatting: V:/fmt-test/src\file1.ts, V:/fmt-test/src\file2.ts', cli\fmt.rs:223:5
```

Reply:
Ok, I was just experimenting with trying out catching panic's in dprint then panicking with the file path. It simplifies the code here a lot because I can switch back to `try_join_all`. The problem is that when multiple panics happen at the same time, the output will start to get merged together. It seems like there's no locking around this, but it's something that will probably happen very rarely.

Anyway, thoughts on doing this instead? Here's what the output would look like:

> thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', V:\dprint\packages\rust-dprint-plugin-typescript\src\parser.rs:1299:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'tokio-runtime-worker' panicked at 'dprint panic formatting: 'V:/fmt-test/src\file1.ts'
Please report this at https://github.com/dsherret/dprint -- For now add a `// dprint-ignore-file` comment at the top of the file.', V:\dprint\packages\rust-dprint-plugin-typescript\src\formatter.rs:74:23thread '
tokio-runtime-worker' panicked at 'not implemented: reparse_expr_as_pat, pat_ty = BindingPat, expr = Arrow(ArrowExpr { span: Span { lo: BytePos(1276817), hi: BytePos(1276901), ctxt: #0 }, params: [Ident(Ident { span: Span { lo: BytePos(1276817), hi: BytePos(1276818), ctxt: #0 }, sym: Atom('t' type=inline), type_ann: None, optional: false })], body: Expr(Bin(BinExpr { span: Span { lo: BytePos(1276822), hi: BytePos(1276900), ctxt: #0 }, op: "||", left: Call(CallExpr { span: Span { lo: BytePos(1276822), hi: BytePos(1276854), ctxt: #0 }, callee: Expr(Ident(Ident { span: Span { lo: BytePos(1276822), hi: BytePos(1276840), ctxt: #0 }, sym: Atom('areTypesComparable' type=dynamic), type_ann: None, optional: false })), args: [ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276841), hi: BytePos(1276842), ctxt: #0 }, sym: Atom('t' type=inline), type_ann: None, optional: false }) }, ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276844), hi: BytePos(1276853), ctxt: #0 }, sym: Atom('valueType' type=dynamic), type_ann: None, optional: false }) }], type_args: None }), right: Call(CallExpr { span: Span { lo: BytePos(1276t858), hi: BytePos(1276900), ctxt: #0 }, callee: Expr(Ident(Ident { span: Span { lo: BytePos(1276858), hi: BytePos(1276886), ctxt: #0 }, sym: Atom('isCoercibleUnderDoublhread 'eEquals' type=dynamic), type_ann: None, optional: false })), args: [ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276887), hi: BytePos(1276main888), ctxt: #0 }, sym: Atom('t' type=inline), type_ann: None, optional: false }) }, ExprOrSpread { spread: None, expr: Ident(Ident { span: Span { lo: BytePos(1276890), ' panicked at 'hi: BytePos(1276899), ctxt: #0 }, sym: Atom('valueType' type=dynamic), type_ann: None, optional: false }) }], type_args: None }) })), is_async: false, is_generator: falcalled `Result::unwrap()` on an `Err` value: JoinError::Panic(...)se, type_params: None, return_type: None })', ', cli\fmt.rs<::std::macros:::panic macros>208::524:
6
thread 'tokio-runtime-worker' panicked at 'dprint panic formatting: 'V:/fmt-test/src\file2.ts'
Please report this at https://github.com/dsherret/dprint -- For now add a `// dprint-ignore-file` comment at the top of the file.', V:\dprint\packages\rust-dprint-plugin-typescript\src\formatter.rs:74:23